nvflare/app_common/executors/launcher_executor.py

```diff
@@ -161,67 +154,81 @@ def execute(self, task_name: str, shareable: Shareable, fl_ctx: FLContext, abort
 
         # if not launched yet
         if not self._launch_once or not self._launched:
-            self._init_pipe_handler()
             self.prepare_config_for_launch(shareable, fl_ctx)
             launch_success = self._launch(task_name, shareable, fl_ctx, abort_signal)
             if not launch_success:
                 self.log_error(fl_ctx, f"launch task ({task_name}): failed")
                 return make_reply(ReturnCode.EXECUTION_EXCEPTION)
             self._launched = True
             self.log_info(fl_ctx, f"External process for task ({task_name}) is launched.")
-
-        if self.pipe_handler is None:
-            self.log_error(fl_ctx, "pipe_handler is None")
-            return make_reply(ReturnCode.EXECUTION_EXCEPTION)
+            self._check_last_result_transfer_thread = threading.Thread(
+                target=self._check_last_result_transfer,
+                args=(
+                    fl_ctx,
+                    abort_signal,
+                ),
+            )
+            self._check_last_result_transfer_thread.start()
 
         # pipe handler starts checking for heartbeats only after the 3rd party code has been launched
         self.pipe_handler.start()
```

Comment:
You shouldn't directly access the "pipe_handler", which is part of the TaskExchanger. 
This pipe_handler should be started by the TaskExchanger already, no?

Reply:
removed this line.